pallets/nfts/src/features/approvals.rs

```diff
@@ -165,10 +177,231 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
+	///   collection or that the approval is past its deadline, granting permission to cancel the
+	///   approval. If `None`, no permission check is performed.
+	/// - `collection`: The identifier of the collection
+	/// - `delegate`: The account that was previously allowed to take control of all items within
+	///   the collection.
+	pub(crate) fn do_cancel_collection(
+		maybe_check_origin: Option<T::AccountId>,
+		collection: T::CollectionId,
+		delegate: T::AccountId,
+	) -> DispatchResult {
+		let owner =
+			Pallet::<T, I>::collection_owner(collection).ok_or(Error::<T, I>::UnknownCollection)?;
+
+		let maybe_deadline = CollectionApprovals::<T, I>::get((&collection, &owner, &delegate))
+			.ok_or(Error::<T, I>::NotDelegate)?;
+
+		let is_past_deadline = if let Some(deadline) = maybe_deadline {
+			let now = frame_system::Pallet::<T>::block_number();
+			now > deadline
+		} else {
+			false
+		};
+
+		if !is_past_deadline {
+			if let Some(check_origin) = maybe_check_origin {
+				ensure!(check_origin == owner, Error::<T, I>::NoPermission);
+			}
+		}
+
+		CollectionApprovals::<T, I>::remove((&collection, &owner, &delegate));
+		AccountApprovals::<T, I>::mutate(collection, &owner, |approvals| {
+			approvals.saturating_dec();
+		});
+
+		Self::deposit_event(Event::ApprovalCancelled { collection, owner, item: None, delegate });
+
+		Ok(())
+	}
+
+	/// Clears all collection approvals.
+	///
+	/// This function is used to clear all approvals to transfer all items within the collections.
+	/// If `maybe_check_origin` is specified, the function ensures that the `check_origin` account
+	/// is the owner of the item, granting permission to clear all collection approvals. After
+	/// clearing all approvals, the function emits the `AllApprovalsCancelled` event.
+	///
+	/// - `maybe_check_origin`: The optional account that is required to be the owner of the
+	///   collection, granting permission to clear all collection approvals. If `None`, no
+	///   permission check is performed.
+	/// - `collection`: The collection ID containing the item.
+	/// - `witness_approvals`: Information on the collection approvals cleared. This must be
+	///   correct.
+	pub(crate) fn do_clear_all_collection_approvals(
+		maybe_check_origin: Option<T::AccountId>,
+		collection: T::CollectionId,
+		witness_approvals: u32,
+	) -> DispatchResult {
+		let owner =
+			Pallet::<T, I>::collection_owner(collection).ok_or(Error::<T, I>::UnknownCollection)?;
+
+		if let Some(check_origin) = maybe_check_origin {
+			ensure!(check_origin == owner.clone(), Error::<T, I>::NoPermission);
+		}
+
+		AccountApprovals::<T, I>::try_mutate(
+			collection,
+			&owner,
+			|approvals| -> Result<(), DispatchError> {
+				ensure!(*approvals == witness_approvals, Error::<T, I>::BadWitness);
+				let _ = CollectionApprovals::<T, I>::clear_prefix(
+					(collection, owner.clone()),
+					*approvals,
+					None,
+				);
+				*approvals = 0;
```

Comment:
Does this remove the value from storage, or set it back with a zero value. We would want the key/value removed, so would be good to ensure this with a test. (I have not checked the tests yet).